src/control/LatexController.cpp

```diff
@@ -166,22 +169,28 @@ void LatexController::handleTexChanged(GtkTextBuffer* buffer, LatexController* s
 }
 
 void LatexController::onPdfRenderComplete(GObject* procObj, GAsyncResult* res, LatexController* self) {
-    g_assert(self->isUpdating);
+    if (!self->isUpdating()) {
+        return;
+    }
```

Comment:
The destructor is being called before the callback is invoked here, so `self` has technically already been deallocated. The check here therefore should not access any members of `self`.

One clean alternative is to pass in a pair of `LatexController*` and `GCancellable*` to `g_subprocess_wait_check_async` and do the cancel check here.

Reply:
I don't think I even need that check now, since the canceled error will be thrown on the previous render attempt, so only one render at a time is running, right?